controllers/storagecluster/cephrbdmirrors.go

```diff
@@ -0,0 +1,136 @@
+)
+
+type ocsCephRbdMirrors struct{}
+
+func (r *StorageClusterReconciler) fetchCephRbdMirrorInstance(cephRbdMirror *cephv1.CephRBDMirror) (*cephv1.CephRBDMirror, error) {
+	existing := &cephv1.CephRBDMirror{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{Name: cephRbdMirror.Name, Namespace: cephRbdMirror.Namespace}, existing)
+	return existing, err
+}
+
+func (r *StorageClusterReconciler) deleteCephRbdMirrorInstance(cephRbdMirrors []*cephv1.CephRBDMirror, flowName string) error {
+	for _, cephRbdMirror := range cephRbdMirrors {
+		existing, err := r.fetchCephRbdMirrorInstance(cephRbdMirror)
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				return fmt.Errorf("%s: unable to retrieve CephRbdMirror %v: %v", flowName, cephRbdMirror.Name, err)
+			}
+			continue
+		}
+		if cephRbdMirror.GetDeletionTimestamp().IsZero() {
+			r.Log.Info(flowName+": Deleting CephRbdMirror.", "CephRbdMirror", klog.KRef(cephRbdMirror.Namespace, cephRbdMirror.Name))
+			err := r.Client.Delete(context.TODO(), existing)
+			if err != nil {
+				r.Log.Info(flowName+": Retrying deletion of CephRbdMirror.", "CephRbdMirror", klog.KRef(existing.Namespace, existing.Name))
+				err := r.Client.Delete(context.TODO(), existing)
+				if err != nil {
+					r.Log.Error(err, flowName+": Failed to delete CephRbdMirror.", "CephRbdMirror", klog.KRef(existing.Namespace, existing.Name))
+					return fmt.Errorf("%s: Failed to delete CephRbdMirror %v: %v", flowName, existing.Name, err)
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// newCephRbdMirrorInstances returns the cephRbdMirror instances that should be created
+// on first run.
+func (r *StorageClusterReconciler) newCephRbdMirrorInstances(initData *ocsv1.StorageCluster) ([]*cephv1.CephRBDMirror, error) {
+	ret := []*cephv1.CephRBDMirror{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      generateNameForCephRbdMirror(initData),
+				Namespace: initData.Namespace,
+			},
+			Spec: cephv1.RBDMirroringSpec{
+				Count:     1,
+				Resources: defaults.DaemonResources["rbd-mirror"],
+			},
+		},
+	}
+	for _, obj := range ret {
+		err := controllerutil.SetControllerReference(initData, obj, r.Scheme)
+		if err != nil {
+			r.Log.Error(err, "Unable to set controller reference for CephRBDMirror.", "CephRBDMirror", klog.KRef(obj.Namespace, obj.Name))
+			return nil, err
+		}
+	}
+	return ret, nil
+}
+
+// ensureCreated ensures that cephRbdMirror resources exist in the desired state.
+func (obj *ocsCephRbdMirrors) ensureCreated(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) error {
+	reconcileStrategy := ReconcileStrategy(instance.Spec.ManagedResources.CephRbdMirrors.ReconcileStrategy)
+	if reconcileStrategy == ReconcileStrategyIgnore {
+		return nil
+	}
+
+	cephRbdMirrors, err := r.newCephRbdMirrorInstances(instance)
+	if err != nil {
+		return err
+	}
+
+	if !instance.Spec.Mirroring.Enabled {
+		return r.deleteCephRbdMirrorInstance(cephRbdMirrors, "Create")
```

Comment:
Using this Create string to find out from which flow I am triggering the deletion

Reply:
It shouldn't matter. A delete is a delete no matter who triggers or how.
Just add a log message saying mirroring is disabled so deleting rbd mirror.

Reply:
ack